packages/@o3r/extractors/src/core/comparator/metadata-comparator.interface.ts

```diff
@@ -0,0 +1,76 @@
+import type { JsonObject } from '@angular-devkit/core';
+import type { SupportedPackageManagers } from '@o3r/schematics';
+
+/**
+ * Interface of the comparator used to compare 2 different versions of the same metadata file.
+ */
+export interface MetadataComparator<T> {
```

Comment:
missing constraint on `T`.
not sur you support `boolean` as value :D

Reply:
If I add a constraint on T, `like <T extends Metadata>` where `type Metadata= JSONLocalization | ComponentConfigOutput | CssVariable`, I will add a dependency on @o3r/localization, @o3r/styling and @o3r/component, which I don't want (and woudn't work)
I think nothing is stopping the used from implementing this interface with any type they want.